java/src/main/java/com/logicalclocks/hsfs/engine/SparkEngine.java

```diff
@@ -303,16 +305,31 @@ private Dataset<Row>[] timeSeriesSplit(TrainingDataset trainingDataset, Query qu
     int i = 0;
     for (Split split : splits) {
       if (dataset.count() > 0) {
+        String eventTime = query.getLeftFeatureGroup().getEventTime();
         String eventTimeType =
-            query.getLeftFeatureGroup().getFeature(query.getLeftFeatureGroup().getEventTime()).getType();
+            query.getLeftFeatureGroup().getFeature(eventTime).getType();
+
         if (BIGINT.getType().equals(eventTimeType)) {
+          sparkSession.sqlContext()
+              .udf()
+              .register("checkEpochUDF", (Long input) -> {
+                if (Long.toString(input).length() == 13) {
+                  input = input / 1000;
+                  return input.longValue();
+                } else {
+                  return input;
+                }
+              }, DataTypes.LongType);
+          dataset = dataset.withColumn(eventTime,functions.callUDF(
```

Comment:
We should use a different name as the new column an drop it later, other this will change the value of the original event time column.